richreview/public/webapps/richreview/r2/app.js

```diff
@@ -21,6 +21,14 @@
                     r2App.cur_page.Relayout();
                     r2App.invalidate_page_layout = false;
                     console.log('invalidate_page_layout');
+                    var cur_view = document.getElementById("r2_view");
```

Comment:
r2_view is already captured in r2.dom's 'view' attribute. Please create pub.getView() in r2.dom, and use r2.dom.getView() instead.